### PR TITLE
[API-39073] Allow brackets in disabilities.name

### PR DESCRIPTION
**Pull request by tycol7**
## Summary

Allow brackets in `disabilities.name`.

## Related issue(s)

[API-39073](https://jira.devops.va.gov/browse/API-39073)

## Testing done

- [x] *New code is covered by unit tests*
- Successfully tested the 526 endpoint with brackets in `disabilities.name`.

## Screenshots
brackets-valid

## What areas of the site does it impact?
526 schema

## Acceptance criteria

- [x]  I updated unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

N/A

### CODE REVIEW

File: modules/claims_api/config/schemas/v2/526.json [hunk 1]
```diff
@@ -543,7 +543,7 @@
           "name": {
             "description": "Name of the disability. The /disabilities endpoint on the [Benefits Reference Data API](https://developer.va.gov/explore/benefits/docs/benefits_reference_data?version=current) may be used to retrieve possible disability names.",
             "type": "string",
-            "pattern": "^$|([a-zA-Z0-9\"\\/&\\(\\)\\-'.,# ]([a-zA-Z0-9(\\)\\-'.,# ])?)+$",
+            "pattern": "^$|([a-zA-Z0-9\"\\/&\\(\\)\\-'.,#\\[\\] ]([a-zA-Z0-9(\\)\\-'.,#\\[\\] ])?)+$",
```
Comment (rockwellwindsor-va) [inline modules/claims_api/config/schemas/v2/526.json:546]: @tycol7 It appears the Docker Container does not accept the name with brackets so I believe the `disability_compensation_evss_mapper` will also need to be updated to remove the brackets from the name.
  
Screenshot 2024-08-15 at 9 28 28 AM

When I removed the brackets via the mapper using `disability[:name] = disability[:name].gsub(/\[|\]/, '')
` the claim established as expected.

Comment (tycol7) [reply modules/claims_api/config/schemas/v2/526.json:546]: https://github.com/department-of-veterans-affairs/vets-api/pull/18037/commits/4c4af5887e87492cd5d4764540b53ff94761827b

**Review by ericboehs — APPROVED** [review]

File: modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb [hunk 2]
```diff
@@ -65,10 +65,14 @@ def transform_disability_values!(disability)
         disability.delete(:serviceRelevance) if disability&.dig(:serviceRelevance).blank?
         disability.delete(:classificationCode) if disability&.dig(:classificationCode).nil? # blank is ok
 
+        disability[:name] = disability[:name].gsub(/\[|\]/, '') if disability[:name].present?
+
         if disability&.dig(:secondaryDisabilities).present?
           disability[:secondaryDisabilities] = disability[:secondaryDisabilities]&.map do |secondary|
             secondary.delete(:classificationCode) if secondary&.dig(:classificationCode).nil? # blank is ok
 
+            secondary[:name] = secondary[:name].gsub(/\[|\]/, '') if secondary[:name].present?
```
Comment (FonzMP) [inline modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb:74]: think this is currently confirming the need for the strip here. EVSS Docker should be updated to handle `[]` now.

Comment (tycol7) [reply modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb:74]: Per Slack convo, we are still getting an error from EVSS when forwarding brackets. Plan is to strip brackets for now and remove these lines after confirming that EVSS accepts brackets.

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**